interface/src/app/node/node.js

```diff
@@ -0,0 +1,98 @@
+/**
+ * "adage.gene.network" module.
+ */
+
+angular.module('adage.node', [
+  'ui.router',
+  'placeholders',
+  'ui.bootstrap'
+])
+
+.config(function config($stateProvider) {
+  $stateProvider.state('node', {
+    url: '/node/{id:int}',
+    views: {
+      main: {
+        templateUrl: 'node/node.tpl.html',
+        controller: 'NodeCtrl as ctrl'
+      }
+    },
+    data: {pageTitle: 'Node Information'}
+  });
+})
+
+.factory('NodeInfo', ['$resource', '$stateParams',
+  function($resource) {
+    return $resource('/api/v0/node/:id');
+  }
+])
+
+.factory('HeavyGenes', ['$resource', function($resource) {
+  return $resource('/api/v0/participation/');
+}])
+
+.controller('NodeCtrl', ['NodeInfo', '$stateParams', '$log',
+  function NodeController(NodeInfo, $stateParams, $log) {
+    var self = this;
+    if (!$stateParams.id) {
+      self.statusMessage = 'Please specify node ID in the URL.';
+      return;
+    }
+    self.id = $stateParams.id;
```

Comment:
I am pondering the use of `self` vs. `this` here. I know you need a closure variable in the callbacks below, but wouldn't it be slightly more clear to use `this` in a spot like this line and a more descriptive closure variable that is clearly labeled for callbacks where needed?

This is mainly a question because I haven't formed a firm opinion yet.